python/nav/web/api/v1/views.py

```diff
@@ -1342,30 +1350,9 @@ class JWTRefreshViewSet(NAVAPIMixin, APIView):
     def post(self, request):
         if not LOCAL_JWT_IS_CONFIGURED:
             return Response("Invalid token", status=status.HTTP_403_FORBIDDEN)
-        # This adds support for requests via the browseable API.
-        # Browseble API sends QueryDict with _content key.
-        # Tests send QueryDict without _content key so it can be treated
-        # as a regular dict.
-        if isinstance(request.data, QueryDict) and '_content' in request.data:
-            json_string = request.data.get('_content')
-            if not json_string:
-                return Response("Empty JSON body", status=status.HTTP_400_BAD_REQUEST)
-            try:
-                data = json.loads(json_string)
-            except json.JSONDecodeError:
-                return Response("Invalid JSON", status=status.HTTP_400_BAD_REQUEST)
-            if not isinstance(data, dict):
-                return Response(
-                    "Invalid request body. Must be a JSON dict",
-                    status=status.HTTP_400_BAD_REQUEST,
-                )
-        elif isinstance(request.data, dict):
-            data = request.data
-        else:
-            return Response(
-                "Invalid request body. Must be a JSON dict",
-                status=status.HTTP_400_BAD_REQUEST,
-            )
+        data, error = _validate_post_data(request.data)
+        if error:
+            return Response(error, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
How are these changes in `JWTRefreshViewSet` relevant to this PR?

Refactoring is fine - but I then prefer that it this happens in a separate commit that explains *why* this was done (it's a lot more helpful when digging back into history with git blame when you can see clearly that "this line of code was changed because of X")